core/src/test/java/com/graphhopper/routing/ch/NodeBasedNodeContractorTest.java

```diff
@@ -502,10 +503,17 @@ private void checkNoShortcuts(CHGraph chGraph) {
     private Shortcut expectedShortcut(int baseNode, int adjNode, EdgeIteratorState edge1, EdgeIteratorState edge2,
                                       boolean fwd, boolean bwd) {
         //todo: weight calculation might have to be adjusted for different encoders/weightings/reverse speed
-        double weight = weighting.calcWeight(edge1, false, EdgeIterator.NO_EDGE) +
-                weighting.calcWeight(edge2, false, EdgeIterator.NO_EDGE);
-        double distance = edge1.getDistance() + edge2.getDistance();
-        return new Shortcut(baseNode, adjNode, weight, distance, fwd, bwd, edge1.getEdge(), edge2.getEdge());
+        double weight1 = getWeight(edge1);
+        double weight2 = getWeight(edge2);
+        return new Shortcut(baseNode, adjNode, weight1 + weight2, fwd, bwd, edge1.getEdge(), edge2.getEdge());
+    }
+
+    private double getWeight(EdgeIteratorState edge) {
+        if (edge instanceof CHEdgeIteratorState) {
+            return ((CHEdgeIteratorState) edge).getWeight();
+        } else {
+            return weighting.calcWeight(edge, false, EdgeIterator.NO_EDGE);
+        }
```

Comment:
ah this is just a helper method in this test, only for convenience. previously I used `weighting.calcWeight` but since weighting is a shortest weighting here this is calling `getDistance()` (which causes an exception for shortcuts). In production code this is a bit different because we normally would use `PreparationWeighting` (but that does not have the instanceof check...).

Reply:
ok